From 269b1a82e654d9b8d682d861b4d0db0bd50d3106 Mon Sep 17 00:00:00 2001
From: Robert Yang <liezhi.yang@windriver.com>
Date: Fri, 27 Nov 2015 03:09:07 +0000
Subject: [PATCH] wtf/Platform.h: fix build for mips n32

If _ABIN32 is defined, it should go into 32 bit branch.

Fixed:
./wtf/StdLibExtras.h: In instantiation of 'TO WTF::bitwise_cast(FROM) [with TO = int; FROM = double]':
runtime/JSValueInlineMethods.h:495:44:   required from here
./wtf/Assertions.h:326:47: error: size of array is negative
 #define COMPILE_ASSERT(exp, name) typedef int dummy##name [(exp) ? 1 : -1]

https://bugreports.qt.io/browse/QTBUG-39224

Upstream-Status: Submitted

Signed-off-by: Robert Yang <liezhi.yang@windriver.com>
---
 src/3rdparty/webkit/Source/JavaScriptCore/wtf/Platform.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/3rdparty/webkit/Source/JavaScriptCore/wtf/Platform.h b/src/3rdparty/webkit/Source/JavaScriptCore/wtf/Platform.h
index e8b03be..dfc6a74 100644
--- a/src/3rdparty/webkit/Source/JavaScriptCore/wtf/Platform.h
+++ b/src/3rdparty/webkit/Source/JavaScriptCore/wtf/Platform.h
@@ -147,7 +147,7 @@
 /* CPU(MIPS) - MIPS 32-bit and 64-bit */
 #if (defined(mips) || defined(__mips__) || defined(MIPS) || defined(_MIPS_) \
     || defined(__mips64))
-#if defined(__mips64)
+#if defined(__mips64) && !defined(_ABIN32)
 #define WTF_CPU_MIPS64 1
 #define WTF_MIPS_ARCH __mips64
 #else
-- 
2.5.0

